srv6: Some cleanups for SRv6 maps

YutaroHayakawa opened the pull request:
Please see individual commits for more details.

```release-note
srv6: Some cleanups for SRv6 maps
```

YutaroHayakawa committed:
srv6: Remove state maps

SRv6 State maps is not used anymore. Remove it from the code base and
implement a cleanup logic on startup. We can remove this cleanup logic
in v1.18.

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

YutaroHayakawa committed:
srv6map: Eliminate global variables

- Eliminate global map variables and rely on the Hive/Cell-based
  initialization.
- Eliminate global datapath macro define and rely on the Hive/Cell-based
  macro injection.
- Eliminate some helpers like Update/Delete since similar functions are
  implemented with bpf package.
- Add tests that ensures the maps and datapath macros are injected as
  intended.

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

YutaroHayakawa committed:
srv6map: Use netip package

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

rgo3 reviewed (approved):
ACK for Loader changes

glibsm reviewed (approved):
Proto change looks good.

derailed reviewed (approved):
@YutaroHayakawa Nice work!

ti-mo commented on pkg/maps/srv6map/policy.go:
```diff
@@ -182,3 +129,71 @@ func (m srv6PolicyMap) IterateWithCallback6(cb SRv6PolicyIterateCallback) error
+
+// OpenPolicyMaps opens the SRv6 policy maps on bpffs
+func OpenPolicyMaps() (*PolicyMap4, *PolicyMap4, error) {
+	m4, err := ebpf.LoadRegisterMap(policyMapName4)
```
Nit: I know this isn't advertised anywhere prominently, but the internal ebpf package is deprecated. Unless there's a feature missing from package bpf, try to use that one instead. Sorry!

YutaroHayakawa replied:
Oops, ok let me take this opportunity and fix that.

YutaroHayakawa replied:
Reflected the suggestion.

YutaroHayakawa committed:
srv6: Add unit tests for OpenXXXMaps and IterateWithCallback

Add unit tests for IterateWithCallback functions and OpenXXXMaps for
Policy, VRF, and SID maps.

Signed-off-by: Yutaro Hayakawa <yutaro.hayakawa@isovalent.com>

julianwiedmann commented on pkg/maps/srv6map/state.go:
```diff
@@ -4,119 +4,18 @@
+// We can delete this in v1.18
+func cleanupStateMap() {
+	os.Remove(bpf.MapPath(StateMapName4))
+	os.Remove(bpf.MapPath(StateMapName6))
```
@YutaroHayakawa :wave: in case you still want to remove this

YutaroHayakawa replied:
Thanks! Will remove this in this PR: https://github.com/cilium/cilium/pull/42184.
